src/main.rs

```diff
@@ -67,6 +68,30 @@ fn notmain() -> Result<i32, anyhow::Error> {
     let bytes = fs::read(elf_path)?;
     let elf = ElfFile::new(&bytes).map_err(|s| anyhow!("{}", s))?;
 
+    let target = probe_rs::config::registry::get_target_by_name(chip)?;
+
+    let mut ram_region = None;
+    for region in &target.memory_map {
+        match region {
+            MemoryRegion::Ram(ram) => {
+                if let Some(old) = &ram_region {
+                    log::debug!("multiple RAM regions found ({:?} and {:?}), stack canary will not be available", old, ram);
```

Comment:
the device may have two regions but the application may only use one; this logic will exclude that kind of applications
this is ok for now but should we file a ticket to tweak the logic further to allow those apps? (not sure what that logic would look like though)

Reply:
Agreed. I've filed https://github.com/knurling-rs/probe-run/issues/39.